### PR TITLE
[Reviewer: Matt] Port the alarm agent to Centos

### PR DESCRIPTION
**Pull request by AlexHockey**
Hi Matt, 

Another PR, this time porting the alarm agent to centos. Again, I've tested that the RPM builds successfully, but I've not done any live testing yet. 

Thanks,
Alex.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,17 +9,25 @@ PREFIX ?= ${ROOT}/usr
...
 
-DEB_COMPONENT := clearwater-snmp-handlers
-DEB_MAJOR_VERSION := 1.0${DEB_VERSION_QUALIFIER}
+PKG_COMPONENT := clearwater-snmp-handlers
+PKG_MAJOR_VERSION ?= 1.0
+
+# Define DEB_NAMES and RPM_NAMES separately as we don't have RPMs for all the
+# components yet.
 DEB_NAMES := clearwater-snmp-handler-cdiv clearwater-snmp-alarm-agent clearwater-snmp-handler-memento-as clearwater-snmp-handler-memento clearwater-snmp-handler-astaire clearwater-snmp-alarm-agent-dbg
+RPM_NAMES := clearwater-snmp-alarm-agent clearwater-snmp-alarm-agent-debuginfo
```
Comment (mirw) [inline Makefile]: Should we changing DEB_NAMES to PKG_NAMES and make cw-rpm.mk be smart about "-dbg" -> "-debuginfo" (or automatically look for "-dbg"/"-debuginfo" versions of every package, as they're always going to be tied to the same repository as their parent)?

Comment (mirw) [inline Makefile]: This would also avoid the need for the RPM_SPECS line below.

File: Makefile [hunk 2]
```diff
@@ -68,9 +76,12 @@ distclean: $(patsubst %, %_distclean, ${SUBMODULES}) cw_alarm_agent_distclean
...
 include build-infra/cw-deb.mk
-
 .PHONY: deb
 deb: build deb-only
 
+include build-infra/cw-rpm.mk
+.PHONY: rpm
+rpm: build rpm-only
+
 .PHONY: all build test clean distclean
```
Comment (mirw) [inline Makefile]: I think our `.PHONY` statements here are a bit broken - `all` doesn't seem to be a target, and above we have `cdiv_handler.so`, which is a target but definitely not phony.  Would you mind taking a quick look?

Comment (mirw) [inline Makefile]: (Just to be clear, nothing to do with your changes.)

File: Makefile (hunk 1 above)
Comment (AlexHockey) [inline Makefile]: I didn't do this originally because it seems like it would couple rpms to deb, but on further thought I'm not sure it does. I think I'd implement this as: 

* `RPM_NAMES` defaults to `PKG_NAMES` but with the `-dbg` -> `-debuginfo` substitution applied. If you ever want to specify `RPM_NAMES` explicitly you have to specify debug packages using `-debuginfo`
* `cw-rpm.mk` spots packages ending in `-debuginfo` and doesn't try to build them. 

Sound OK?

File: Makefile (hunk 2 above)
Comment (AlexHockey) [inline Makefile]: Will do.

File: debian/clearwater-snmp-alarm-agent.upstart [hunk 3]
```diff
@@ -1,37 +1,13 @@
...
-# running in different namespaces to obtain sockets in other namespaces.
-
 description "Clearwater alarm agent"
 
 start on (starting clearwater-monit)
 stop on runlevel [!2345]
 
 respawn
-# Sleep for 5 seconds on stop, to avoid respawning too quickly
-post-stop exec sleep 5
```
Comment (mirw) [inline debian/clearwater-snmp-alarm-agent.upstart]: Why don't we need this any more?

File: rpm/clearwater-snmp-alarm-agent.spec [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+systemctl enable clearwater-snmp-alarm-agent
+systemctl start clearwater-snmp-alarm-agent
+
+%preun
+systemctl stop clearwater-snmp-alarm-agent
+systemctl disable clearwater-snmp-alarm-agent
+
+%postun
+# Trigger an snmpd restart to unload the sub-agent
+systemctl stop snmpd || true
```
Comment (mirw) [inline rpm/clearwater-snmp-alarm-agent.spec]: Do we actually need to do this?  We would for .so files, but I'm surprised for AgentX processes.

**Comment by mirw** [comment]
Looks good.  Comments inline.

File: Makefile (hunk 1 above)
Comment (mirw) [inline Makefile]: As discussed, I think it would be cleaner to

*   remove `-dbg` and `-debuginfo` from `PKG_NAMES`, `DEB_NAMES` and `RPM_NAMES`
*   in cw-rpm.mk,

    *    build the spec files for all `RPM_NAMES`
    *    copy all packages named by `RPM_NAMES` or named by `RPM_NAMES` with a `-debuginfo` suffix

*   in cw-deb.mk, copy all packages named by `DEB_NAMES` or named by `DEB_NAMES` with a `-dbg` suffix
*   remove `RPM_SPECS` entirely (as it's not required).

Sound reasonable?

Comment (AlexHockey) [inline Makefile]: Yep. I have implemented this but want to do some more testing before sending for re-review.

File: debian/clearwater-snmp-alarm-agent.upstart (hunk 3 above)
Comment (AlexHockey) [inline debian/clearwater-snmp-alarm-agent.upstart]: I deleted this and moved the code into the script block below (for consistency with other scripts). Though in retrospect this way seems better (since you sleep after the process has exited, rather than before it starts). I'll migrate the other script I saw to use this approach.

File: rpm/clearwater-snmp-alarm-agent.spec (hunk 4 above)
Comment (AlexHockey) [inline rpm/clearwater-snmp-alarm-agent.spec]: I'm not really sure, but:

* It's only called on post uninstall, which will rarely be executed in practice. 
* Changing this behaviour seems out of scope of porting to centos.

Comment (mirw) [inline rpm/clearwater-snmp-alarm-agent.spec]: Fine.

File: Makefile (hunk 2 above)
Comment (AlexHockey) [inline Makefile]: It turns out that `all` is a target (defined at the top of the file) and the other targets that look like they shouldn't be phony are actually phonies.
